### DISCUSSION TITLE
Can terminal send associtated text for Enter/Tab/Backspace keys?

### DISCUSSION THREAD

**Discussion by kchibisov** [kovidgoyal/kitty#7050] [Q&A] [answered]
I'm not sure why it shouldn't given that libxkbcommon usually attaches something and some platforms do so as well.

In general, I'm not sure how it can break clients, however it was brought to me that only alacritty does that (we just forward text that we were given from the system library).

If it must not be sent, the spec should explicitly say so and also other cases where there's  associated text must not be emitted.

cc @rockorager

**Comment by rockorager** [comment, marked as answer]
Reference: https://github.com/kovidgoyal/kitty/issues/7051

Feel free to close the issue if discussions is the best place for this.

**Comment by rockorager** [comment]
Re: how it can break clients:

Some simple logic for a text input would be
```
if key_event.has_text { 
    insert_text(text) 
}
if key_event.is_backspace {
    delete_character()
}
```

It's easy enough to swap the logic here, but it's also unexpected that a backspace produces text I should print
